jobs/transforms/exp_validation.py

```diff
@@ -74,19 +75,25 @@ class ValidateExplorationCommitLogEntryModel(
     """Overrides _get_change_domain_class for exploration models """
 
     def _get_change_domain_class(self, input_model):
-        """Returns a Change domain class.
+        """Returns a change domain class.
 
         Args:
             input_model: datastore_services.Model. Entity to validate.
 
         Returns:
-            change_domain.BaseChange. A domain object class for the
-            changes made by commit commands of the model.
+            ExplorationRightsChange|ExplorationChange. A domain object class for
+            the changes made by commit commands of the model.
+
+        Raises:
+            Exception. Entity id does not match regex pattern.
         """
         model = job_utils.clone_model(input_model)
         if model.id.startswith('rights'):
             return rights_domain.ExplorationRightsChange
         elif model.id.startswith('exploration'):
             return exp_domain.ExplorationChange
         else:
-            return None
+            raise Exception(
+                'model %s Entity id %s: Entity id does not match regex '
+                'pattern' % (
+                    base_model_validators.ERROR_CATEGORY_ID_CHECK, model.id))
```

Comment:
Should we introduce new exception type for this?

Reply:
I think that will be a good idea since that is less ambiguos and also I will be raising the same exception for other models. 
I wonder where is a good place to define that exception. I think `base_validation.py` is a good place WDYT ?

Reply:
@vojtechjelinek I have defined the exception in `base_validation_errors.py` instead because all errors are defined there